### PR TITLE
Support storing protoDescriptorBin in a ConfigMap

### PR DESCRIPTION
**Pull request by nfuden**
Configmap support for proto descriptors when using grpcjson plugin. This reduces the size of the gateway object definition which previously could break the rollout job.

To use this in helm, old helm values in the shape of:
```
gatewayProxies:
  gatewayProxy:
    gatewaySettings:
      customHttpGateway:
        options:
          grpcJsonTranscoder:
            protoDescriptorBin: <proto-desc-content...>
```
should be changed to:
```
gatewayProxies:
  gatewayProxy:
    gatewaySettings:
      customHttpGateway:
        options:
          grpcJsonTranscoder:
            protoDescriptorConfigMap:
              configMapRef:
                name: my-config-map
                namespace: gloo-system  # optional, defaults to install namespace
              key: my-key  # optional if there is only one key in the configmap
global:
  configMaps:
  - name: my-config-map
     namespace: gloo-system
     data:
       my-key: <proto-desc-content...>
```

### CODE REVIEW

File: projects/gloo/pkg/plugins/grpcjson/plugin.go [hunk 1]
```diff
@@ -100,35 +118,46 @@ func translateGlooToEnvoyPrintOptions(options *grpc_json.GrpcJsonTranscoder_Prin
...
 }
 
-func translateConfigMapToProtoBin(ctx context.Context, snap *gloosnapshot.ApiSnapshot, configRef *grpc_json.GrpcJsonTranscoder_DescriptorConfigMap) (out []byte) {
+// get the proto descriptor data from a ConfigMap
+func translateConfigMapToProtoBin(ctx context.Context, snap *gloosnapshot.ApiSnapshot, configRef *grpc_json.GrpcJsonTranscoder_DescriptorConfigMap) ([]byte, error) {
+	if configRef.GetConfigMapRef() == nil {
+		return nil, NoConfigMapRefError()
+	}
 
+	// make sure the referenced configmap exists and has data
```
Comment (nfuden) [inline projects/gloo/pkg/plugins/grpcjson/plugin.go:127]: Nit:Exists in gloo snapshot

**Comment by jenshu** [comment]
undrafting so ci runs

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -552,10 +553,6 @@ type AccessLogger struct {
 	*DeploymentSpec
 }
 
-type GatewayProxyConfigMap struct {
```
Comment (jenshu) [inline install/helm/gloo/generate/values.go:555]: this struct was not used/referenced anywhere

File: test/e2e/grpc_json_test.go [hunk 3]
```diff
@@ -92,69 +89,119 @@ var _ = Describe("GRPC to JSON Transcoding Plugin - Envoy API", func() {
 		}
 	}
 
-	It("Routes to GRPC Functions", func() {
+	testRequest := func(path string, shouldMatch bool) {
+		body := []byte(`"foo"`) // this is valid JSON because of the quotes
```
Comment (elcasteel) [inline test/e2e/grpc_json_test.go:93]: nit: `"foo"` could be a constant that we use in all the comparisons.

Comment (elcasteel) [reply test/e2e/grpc_json_test.go:93]: Although now I see that this is moved rather than new so even less pressing.

Comment (jenshu) [reply test/e2e/grpc_json_test.go:94]: 2a2e564

**Review by jenshu — APPROVED** [review]
approving on behalf of @nfuden

**Pull request merged**